globals.go

```diff
@@ -1084,3 +1084,55 @@ const (
 	LogViewerHTML    LogViewer = "html"
 	LogViewerLobster LogViewer = "lobster"
 )
+
+// ContainerOS denotes the operating system of a running container. Only linux and windows are supported.
+type ContainerOS string
+
+const (
+	LinuxOS   = "linux"
+	WindowsOS = "windows"
+)
+
+func (c ContainerOS) Validate() error {
+	switch c {
+	case LinuxOS, WindowsOS:
+		return nil
+	default:
+		return errors.Errorf("unrecognized container OS '%s'", c)
+	}
+}
+
+// CPUArchitecture represents the architecture property of a container required to provide to Amazon ECS
```

Comment:
I would avoid saying "Amazon ECS" here since that assumes a particular container implementation. Instead, I would say something more generic like "CPUArchitecture is the architecture necessary to run this container".